libraries/GCS_MAVLink/GCS_Common.cpp

```diff
@@ -1525,6 +1526,14 @@ void GCS_MAVLINK::send_vibration() const
         ins.get_accel_clip_count(2));
 }
 
+void GCS_MAVLINK::send_named_float(const char *name, float value) const
+{
+    char float_name[MAVLINK_MSG_NAMED_VALUE_FLOAT_FIELD_NAME_LEN+1] {};
```

Comment:
Why the +1? The char array needs exactly MAVLINK_MSG_NAMED_VALUE_FLOAT_FIELD_NAME_LEN. Contrary to vsnprintf, strncpy will copy every character until it can, it doesn't reserve space for null-terminator.

Reply:
GCC 8.1 is killing me. Is it a compiler or a teaching tool? Gosh.

Anyway, rather than disabling the warning I'd prefer to use the strategy that GCC docs suggest (https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html): _To avoid the warning when the result is not expected to be NUL-terminated, call memcpy instead._

The incoming string has to be NUL-terminated, so doing a memcpy should always work, it either sends a NUL-terminated string smaller than the array size or it sends a not NUL-terminated string truncated to the array size.